joern-cli/frontends/php2cpg/src/test/scala/io/joern/php2cpg/querying/MethodTests.scala

```diff
@@ -146,7 +146,7 @@ class MethodTests extends PhpCode2CpgFixture {
         fooMethod.file.head.content.substring(offsetStart, offsetEnd) shouldBe
           """function foo() {
             |  // ⦝
-            |  $x = "🙂⨌🙂𐇐🙂🙂🙂🙂";
+            |  $x = "??⨌????????????";
```

Comment:
Using `IOUtils.readEntireFile` introduces a regression in the characters we can actually parse, but I suspect it's unlikely we'll encounter many such characters in actual code and is outweighed by the benefit of using a uniform approach across frontends

Reply:
Hm, stange. It works here: io.joern.jssrc2cpg.io.CodeDumperFromContentTests

Reply:
I get the same results when pasting that string into one of the jssrc tests:

```
...
  x = 'foo[??⨌????????????]';
}" was not equal to "...ass Foo
{
  x = 'foo[🙂⨌🙂𐇐🙂🙂🙂🙂]';
...
```

I did a hex dump of a file containing only `🙂` and it matches the UTF-8 encoding from https://www.compart.com/en/unicode/U+1F642

```
❯ hexdump /tmp/smile
0000000 9ff0 8299 000a                         
0000005
❯ cat /tmp/smile
🙂
```

nvim could've done some conversions on write or such though. If not, it looks like it should be a valid UTF-8 character

Reply:
Maybe its time to move IOUtils to x2cpg (which wasnt a thing back then) and fix the utf8 handling. I'll do a draft PR.

Reply:
With this: https://github.com/ShiftLeftSecurity/codepropertygraph/pull/1804 the test case works again. 
I was not able to (re-) discover some code with actual unpaired surrogates (which was the reason I implemented it in first place). Not really sure on how to proceed there.

Reply:
does this table here help with crafting such a file? https://simonsapin.github.io/wtf-8/#surrogates-byte-sequences

Reply:
Interesting! We should add some testing for that.